soapbox/python/nicolascarbone.md

```diff
@@ -0,0 +1,111 @@
+# [Nicolas Carbone's](https://github.com/nicolascarbone) Soapbox of Python Practices
+
+This soapbox contains some basic stuff for different Python projects.
+
+## Coding Style
+
+Python has defined a coding style guide called PEP8, lease refer to [PEP8][https://www.python.org/dev/peps/pep-0008/] for a complete reference.
+
+You should always stick to this guide, it is in most of the python projects.
+
+Most important styles are:
+- Use spaces instead of tabs.
+- Use underscode for function, methods, variables and file names:
+    - Examples: get_user, client_name, set_date, etc
+- Use camel case for class names: MyClass, MyService, etc
+- Use capital letter at module level for constants.
+- Use `self` as first argument to instance methods.
+- Use `cls` as first argument to class methods.
+
+### Linting
+
+There are many linters in Python, choose the one you like the most and stick to it.
+Some of them are
+- Pylint
+- Flake8
+- Pylama
+etc
+
+The whole team should use the same linter for sake of consistency.
+
+You can also use git hooks to make sure you don't push any code that does not pass a linter analysis.
+
+### Formatting
+
+There are many formatters in python (you can see most of them [Here][https://github.com/life4/awesome-python-code-formatters] )
+
+Choose the one you like the most but please use at least one.
+
+Again, the whole team should use the same formatter for consistency.
+
+Formatters can run in a git hook or as an IDE plugin.
+
+### Import statements
+
+Keep imports sorted, the following is a very common convention
+
+```
+<python standard lib imporst>
+
+<third party imports>
+
+<app packages>
+
+<tests> (if it is a test suite)
+```
+
+You can always use plugins like [isort][https://pypi.org/project/isort/] to do this when you save a file or in a git hook.
+
+<br>
+
+### Private Class Properties
+
+Python does not have access modifiers so use a leading `_` for private methods.
+
+Avoid using leading `__` because that will trigger a process called `name mangling`, it will just put the classname as attribute preffix.
+
+## Coding Practices
+
+As general rule, your code should have high cohesion and low coupling, this will make it easier to maintain, scale, etc.
+
+Some things to keep in mind:
+- Dependency Injection: This will decouple your services from other ones and it will make them easier to test and maintain.
```

Comment:
Is there a preferred way to do this that you can reference/link?

Reply:
yes, I'll add links to some implementations